modules/xfem/src/efa/EFAFragment2D.C

```diff
@@ -334,355 +334,69 @@ EFAFragment2D::split()
-      EFANode * node2 = _boundary_edges[1]->getNode(0);
-      EFANode * node3 = _boundary_edges[2]->getNode(0);
-      EFANode * node4 = _boundary_edges[3]->getNode(0);
-
-      new_frag1->addEdge(new EFAEdge(node1, node2));
-      new_frag1->addEdge(new EFAEdge(node2, node3));
-      new_frag1->addEdge(new EFAEdge(node3, node1));
-
-      new_frag2->addEdge(new EFAEdge(node3, node4));
-      new_frag2->addEdge(new EFAEdge(node4, node1));
-      new_frag2->addEdge(new EFAEdge(node1, node3));
-
-      new_fragments.push_back(new_frag1);
-      new_fragments.push_back(new_frag2);
-    }
-    else if ((cut_nodes[0] == 1 && cut_nodes[1] == 3) || (cut_nodes[0] == 3 && cut_nodes[1] == 1))
-    {
-      EFAFragment2D * new_frag1 = new EFAFragment2D(_host_elem, false, NULL);
-      EFAFragment2D * new_frag2 = new EFAFragment2D(_host_elem, false, NULL);
-      EFANode * node1 = _boundary_edges[0]->getNode(0);
-      EFANode * node2 = _boundary_edges[1]->getNode(0);
-      EFANode * node3 = _boundary_edges[2]->getNode(0);
-      EFANode * node4 = _boundary_edges[3]->getNode(0);
-
-      new_frag1->addEdge(new EFAEdge(node1, node2));
-      new_frag1->addEdge(new EFAEdge(node2, node4));
-      new_frag1->addEdge(new EFAEdge(node4, node1));
-
-      new_frag2->addEdge(new EFAEdge(node2, node3));
-      new_frag2->addEdge(new EFAEdge(node3, node4));
-      new_frag2->addEdge(new EFAEdge(node4, node2));
-
-      new_fragments.push_back(new_frag1);
-      new_fragments.push_back(new_frag2);
-    }
-    else if ((cut_nodes[0] == 0 && cut_nodes[1] == 1) || (cut_nodes[0] == 1 && cut_nodes[1] == 2) ||
-             (cut_nodes[0] == 2 && cut_nodes[1] == 3) || (cut_nodes[0] == 3 && cut_nodes[1] == 0) ||
-             (cut_nodes[0] == 1 && cut_nodes[1] == 0) || (cut_nodes[0] == 2 && cut_nodes[1] == 1) ||
-             (cut_nodes[0] == 3 && cut_nodes[1] == 2) || (cut_nodes[0] == 0 && cut_nodes[1] == 3))
+    fragment_nodes[frag_number].push_back(
+        _boundary_edges[iedge]->getNode(0)); // add current node to current new frag
+
+    if (_boundary_edges[iedge]->getNode(0)->category() ==
+        EFANode::N_CATEGORY_EMBEDDED_PERMANENT) // if current node has been cut change fragment
     {
-      EFAFragment2D * new_frag = new EFAFragment2D(_host_elem, false, NULL);
-      EFANode * node1 = _boundary_edges[0]->getNode(0);
-      EFANode * node2 = _boundary_edges[1]->getNode(0);
-      EFANode * node3 = _boundary_edges[2]->getNode(0);
-      EFANode * node4 = _boundary_edges[3]->getNode(0);
-
-      new_frag->addEdge(new EFAEdge(node1, node2));
-      new_frag->addEdge(new EFAEdge(node2, node3));
-      new_frag->addEdge(new EFAEdge(node3, node4));
-      new_frag->addEdge(new EFAEdge(node4, node1));
-
-      new_fragments.push_back(new_frag);
-    }
-  }
-  else if (cut_edges.size() == 2 || (cut_edges.size() == 1 && cut_nodes.size() == 0))
-  {
-    unsigned int iedge = 0;
-    unsigned int icutedge = 0;
+      if (frag_number == 0)
+      {
```

Comment:
Our convention is to not use curly braces around one-line blocks. You have a bunch of these -- just delete the opening and closing braces.